sunpy/net/dataretriever/sources/goes.py

```diff
@@ -269,9 +269,9 @@ def search(self, *args, **kwargs):
         matchdict = self._get_match_dict(*args, **kwargs)
         req_wave = matchdict.get('Wavelength', None)
         if req_wave is not None:
-            wmin = req_wave.min.to(u.Angstrom, equivalencies=u.spectral())
-            wmax = req_wave.max.to(u.Angstrom, equivalencies=u.spectral())
-            req_wave = a.Wavelength(wmin, wmax)
+            wmin_round = round(req_wave.min.value) * u.Angstrom
+            wmax_round = round(req_wave.max.value) * u.Angstrom
+            req_wave = a.Wavelength(wmin_round, wmax_round)
```

Comment:
How does this logic account for different wavelength units?

Reply:
since all of the wavelengths are being converted to angstrom in a.wavelength so maybe this might be suboptimal method

Reply:
but if i pass 1 mega meter
```
results = Fido.search(
    a.Time(day_start, day_end),
    a.Instrument.suvi,
    a.Level(2),
    a.goes.SatelliteNumber(18),
    a.Wavelength(1*u.Mm),
)
```

output (and if we print wmax_round  wmin_round)
```
9999999999999998.0 Angstrom (wmax round)
9999999999999998.0 Angstrom (wmin round)

Results from 1 Provider:

0 Results from the SUVIClient:
Source: https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes

<No columns>
```

Reply:
Well it seems to work correctly then, I did not expect that based on how I thought the code would convert the values.